nexus/src/app/sagas/instance_create.rs

```diff
@@ -281,34 +276,51 @@ impl NexusSaga for SagaInstanceCreate {
             ));
             subsaga_append(
                 "external_ip",
-                subsaga_builder,
+                subsaga_builder.build()?,
                 &mut builder,
                 repeat_params,
                 i,
             )?;
         }
 
+        for (i, disk) in params.create_params.disks.iter().enumerate() {
+            if let params::InstanceDiskAttachment::Create(create_disk) = disk {
+                let subsaga_name =
+                    SagaName::new(&format!("instance-create-disk-{i}"));
+                let subsaga_builder = DagBuilder::new(subsaga_name);
+                subsaga_append(
+                    "create-disk",
+                    SagaDiskCreate::make_saga_dag(
+                        &disk_create::Params {
+                            serialized_authn: params.serialized_authn.clone(),
+                            project_id: params.project_id.clone(),
+                            create_params: create_disk.clone(),
+                        },
+                        subsaga_builder,
+                    )?,
+                    &mut builder,
+                    create_disk,
+                    i,
+                )?;
+            }
+        }
```

Comment:
@smklein this is the bit that wires up the disk create saga as a proper subsaga. We're literally just re-creating the dag and sending it through `subsaga_append`. There might be a better way to handle the `disk_create` params.